tests/unit/src/aws_iot_tests_unit_json_utils_helper.c

```diff
@@ -58,26 +60,48 @@ TEST_C(JsonUtils, ParseStringBasic) {
 TEST_C(JsonUtils, ParseStringLongerStringIsValid) {
 	int r;
 	const char *json = "{\"x\":\"this is a longer string for test 2\"}";
-	char parsedString[50];
+	char parsedString[STRING_BUFFER_LENGTH];
 
 	IOT_DEBUG("\n-->Running Json Utils Tests - Parse long string \n");
 
 	r = jsmn_parse(&test_parser, json, strlen(json), t, sizeof(t) / sizeof(t[0]));
-	rc = parseStringValue(parsedString, json, t + 2);
+	rc = parseStringValue(parsedString, STRING_BUFFER_LENGTH, json, t + 2);
 
 	CHECK_EQUAL_C_INT(3, r);
 	CHECK_EQUAL_C_INT(SUCCESS, rc);
 	CHECK_EQUAL_C_STRING("this is a longer string for test 2", parsedString);
 }
 
+/* Test that parsing a string doesn't overflow the given buffer. */
+TEST_C(JsonUtils, ParseStringWithBufferTooSmall) {
```

Comment:
You should test the happy path too :)

Reply:
Like when the buffer is big enough?  I think that's adequately tested by the integration tests (jobs integration test).